homeassistant/components/tradfri/sensor.py

```diff
@@ -48,18 +63,41 @@ def _get_air_quality(device: Device) -> int | None:
     return cast(int, device.air_purifier_control.air_purifiers[0].air_quality)
 
 
-SENSOR_DESCRIPTION_AQI = TradfriSensorEntityDescription(
-    device_class=SensorDeviceClass.AQI,
-    native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
-    key=SensorDeviceClass.AQI,
-    value=_get_air_quality,
+def _get_filter_time_left(device: Device) -> int:
+    """Fetch the filter's remaining life (in hours)."""
+    return cast(
```

Comment:
Do we still need to cast here?

Reply:
I tried removing them before pushing this PR:
```
homeassistant/components/tradfri/sensor.py:63: error: Returning Any from function declared to return "Optional[int]"  [no-any-return]
```

Doesn't quite make sense though as the property is an int as expected:

![image](https://user-images.githubusercontent.com/21142447/152681570-47bf1739-ae14-42ea-9e6f-1996a0f4fd16.png)

Reply:
It may be due to the library not specifying typing is supported with `py.typed` file in the package.

https://www.python.org/dev/peps/pep-0561/#packaging-type-information

Although, the whole library isn't typed yet so we should probably wait until that's done before adding `py.typed`.

Reply:
Noted, let's wait.